### PR TITLE
Add a simple Python Flask example. 

### LINKED ISSUE #54: Add simple Python Flask example
* In `examples/` folder
* See https://github.com/leplatrem/minimal-demo-iam/

### PR DESCRIPTION
**Pull request by Natim**
Fixes #54

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Doorman client example
+
+Here are some example of how to integrate Doorman with your service.
```
Comment (leplatrem) [inline examples/README.md]: *examples*

File: examples/python/Pipfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+name = "pypi"
+
+[dev-packages]
+
+[packages]
+
+flask = "*"
+"python-dotenv" = "*"
+"python-jose" = "*"
+"flask-cors" = "*"
```
Comment (leplatrem) [inline examples/python/Pipfile]: trailing blank line

File: examples/python/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Python / Flask example
+
+## How to run the server
+
+    pipenv install
+    API_AUDIENCE="http://minimal-demo-iam.localhost:8000" pipenv run python server.py
```
Comment (leplatrem) [inline examples/python/README.md]: Should we rename it to `API_ID` ? BTW the value is wrong (see `main.js`)

File: examples/python/server.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+from doorman import allowed, AuthZError
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+IAM_SERVER = os.getenv("IAM_SERVER", "http://localhost:8080")
+API_AUDIENCE = os.getenv("API_AUDIENCE")
+
+app = Flask(__name__)
+
```
Comment (leplatrem) [inline examples/python/server.py]: `allowed = functools.partial(DOORMAN_SERVER, API_AUDIENCE)` ?

File: examples/python/server.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            _app_ctx_stack.top.current_user = payload
+            return f(*args, **kwargs)
+        return wrapper
+    return wrapped
+
+
+@app.route("/")
+@cross_origin(headers=["Content-Type", "Authorization"])
+@cross_origin(headers=["Access-Control-Allow-Origin", "*"])
+@authorized(resource="demo:hello", action="read")
```
Comment (leplatrem) [inline examples/python/server.py]: cf review in https://github.com/leplatrem/minimal-demo-iam/pull/6

File: examples/python/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Python / Flask example
+
```
Comment (leplatrem) [inline examples/python/README.md]: Add *How to run Doorman* ? And provide the `policies.yaml`?

File: examples/python/ui/main.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+function main() {
+  const logoutBtn = document.getElementById('logout');
+  logoutBtn.addEventListener('click', logout);
+
+  const webAuth = new auth0.WebAuth({
+    domain: AUTH0_DOMAIN,
+    clientID: AUTH0_CLIENT_ID,
+    redirectUri: AUTH0_CALLBACK_URL,
+    audience: API_AUDIENCE,
```
Comment (leplatrem) [inline examples/python/ui/main.js]: Does this work in the end?

Comment (Natim) [reply examples/python/ui/main.js]: I don't think so but it should be a bug to file in the auth0 lib

**Comment by Natim** [comment]
The current demo is blocked by: `The url "http://doorman.local:3000/" is not in the list of allowed callback URLs: http://iam.local:3000/` ?

File: examples/python/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Python / Flask example
```
[the remark this answers is not in the document]
Comment (Natim) [reply examples/python/README.md]: What do you expect? To list all of them?

File: examples/python/README.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+## How to run doorman?
+
+- Copy the policies.yaml file in the doorman repository.
+- Run `make serve`
+
+
+## How to run the server
+
+    pipenv install
+    API_AUDIENCE="SLocf7Sa1ibd5GNJMMqO539g7cKvWBOI" pipenv run python server.py
```
[the remark this answers is not in the document]
Comment (Natim) [reply examples/python/README.md]: It is the default value.

File: examples/python/server.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    filename = os.path.join(HERE, "records", "{record_id}.json".format(
+        record_id=os.path.basename(record_id)))
+    record = None
+    if os.path.exists(filename):
+        with open(filename, 'r') as f:
+            record = json.load(f)
+            author = record["author"]
+    if request.method == "GET":
+        # READ
+        allowed(resource="record", action="read", jwt=jwt, context={"author": author})
```
[the remark this answers is not in the document]
Comment (Natim) [reply examples/python/server.py]: That's why we had new before :)

**Comment by Natim** [comment]
Should we add unittests?

**Comment by leplatrem** [comment]
> Should we add unittests? 

I wouldn't bother at this stage, and just merge as it is. If you feel motivated, please add an issue to track it :)

**Pull request merged**